### ISSUE TITLE
Request for sample projects built on qcubed v4

### ISSUE THREAD

**Issue by neuroticimbecile** [qcubed/qcubed#1331]
Good day!
I just wanted to ask if any of you would be willing to share some qcubed v3.1 projects (source code and db structure) with an eager learner.  Thank you very much!

**Comment by kukrik** [comment]
Hello! No one (here I may be wrong) is working on v3.1. I have successfully used the QCubed v4.0 project.

Because there has been a great silence for a long time. And PHP versions are also advanced and older versions are no longer supported or will be discontinued.

I recommend switching to QCubed v4. Since there is a long silence here and Composer 2.0 is very much needed, QCubed v4 did not work here anymore.

I cloned my repo https://github.com/qcubed-4 from it and made as few changes to the core of the framework as possible and upgraded to Composer 2.0. Now everything works as it should.

Maybe think about it a bit and switch to QCubed-4? 

You can also view examples from https://qcubed.eu/vendor/qcubed-4/application/assets/php/examples/index.php.

PS. I'm sorry my English is weak :(

**Comment by kukrik** [comment]
To get started, start at https://github.com/qcubed-4/app-starter

**Comment by neuroticimbecile** [comment]
> Hello! No one (here I may be wrong) is working on v3.1. I have successfully used the QCubed v4.0 project.
> 
> Because there has been a great silence for a long time. And PHP versions are also advanced and older versions are no longer supported or will be discontinued.
> 
> I recommend switching to QCubed v4. Since there is a long silence here and Composer 2.0 is very much needed, QCubed v4 did not work here anymore.
> 
> I cloned my repo https://github.com/qcubed-4 from it and made as few changes to the core of the framework as possible and upgraded to Composer 2.0. Now everything works as it should.
> 
> Maybe think about it a bit and switch to QCubed-4?
> 
> You can also view examples from https://qcubed.eu/vendor/qcubed-4/application/assets/php/examples/index.php.
> 
> PS. I'm sorry my English is weak :(

Thank you very much for this info! 
I'll certainly look into it.

How is your experience with QCubed v4?
Does it still need a lot of work to get it beyond "alpha"?
Is it stable enough to use in production?

Anybody out there who wants to share projects (code and db structure) which we can study and learn from?
Thank you!

**Comment by kukrik** [comment]
> How is your experience with QCubed v4?

I was QCodo for a short time, then switched to QCubed from the beginning (from v2 to v4). I'm currently using the latest version, I like it!

> Does it still need a lot of work to get it beyond "alpha"?

I am sorry to say that there has been a long silence here lately. In the past, the community was more active. Why is the latest version given "alpha" status, then the active contribution of the community and several tests are needed to confirm the stable version.

> Is it stable enough to use in production?

I believe this is a stable enough version to use for production. I used to have 2 small projects, then version v2. I don't want to go to this version anymore. I am currently working on one major project based on v4. I have done about 2/3 of the work. This is more of a business logic development topic. Since the necessary plugins were not available for v4, I made the necessary plugins, it took me a lot of time and more tests. Personally, I think I need to add a few more plugins to my project to make it work ... But it still takes time and testing. 

> Anybody out there who wants to share projects (code and db structure) which we can study and learn from?

This last question is a little incomprehensible to me. Could you clarify the question a bit?

Here I'm trying to figure out if you're thinking about how to try and get to know, then there's everything inside QCubed-4.

I don't know if you're using a Windows or Macbook machine. I'm using the last machine. I have no experience with Windows.

The terminal must be used and, for example, type: `cd /usr/local/var/www/htdocs`
Next command: `composer create-project --stability=dev qcubed-4/app-starter your_dir_name`

Then all the necessary things will be downloaded, there is no need to download them separately (instructions can be found at https://github.com/qcubed-4/app-starter)

There is the desired db structure: https://github.com/qcubed-4/application/blob/master/assets/php/examples/mysql_innodb.sql.

After you configure the database settings correctly `/docroot/project/includes/configuration/active/1database.cfg.php.`

Once all the necessary things have been done, you need to go to the home page and run the code generator. I'll take a small screenshot.

<img width="1261" alt="Screenshot 2021-08-05 at 03 42 23" src="https://user-images.githubusercontent.com/2853034/128274097-f7c8f0a2-1af5-457b-ade6-5febfe473e02.png">

It should be remembered here that QCubed Unit Tests are not working yet. Because I had to upgrade PhpUnit version according to the rules of Composer 2.0. I need to rewrite phpunit classes and tests here, I don't have that much free time right now. But this work will have to be done sooner or later.

PS. Previously active, @spekary  did a great job of creating the QCubed v4 branch. Many thanks to him! He's on a new challenge - he's building QCubed GO on the language.

**Comment by neuroticimbecile** [comment]
Thank you again for your reply.
I'm so glad somebody else out there is still using qcubed :-)
I also started with qcodo and then moved to (and got stuck on) qcubed v2.
I am using Linux.

When I said: "Anybody out there who wants to share projects (code and db
structure) which we can study and learn from?" - I was asking the other
members of this group if they would share any of their projects so we can
learn from them.
I am still hoping somebody would.

I am currently trying out the app-starter. Thank you for introducing it!
I encountered some problems. I will just ask about it in the github issues
section of app-starter.

I am also very thankful to @spekary  for taking
qcubed into v4. Mabuhay!

**Comment by neuroticimbecile** [comment]
I was able to run the app-starter!
I was just a bit misled by the error message which was referring to a file called **_config.cfg.php_** which I couldn't find.
I realized that it was actually referring to **_0config.cfg.php_**
Thank you @kukrik !

Now to start building something with it...

![FireShot Capture 023 -  - 11 180 164 25](https://user-images.githubusercontent.com/6220848/128392842-73b6d710-8421-44f4-a89e-bb28dc9d88ba.png)

BTW, I have changed the subject to "...qcubed v4".

**Comment by kukrik** [comment]
Nice to hear! Post any issues with QCubed-4 at https://github.com/qcubed-4/application/issues. Then it is understandable that one or the other problem or issue is related to QCubed-4.

I think the problem with 0config.cfg.php is this line: https://github.com/qcubed-4/application/blob/master/install/project/includes/configuration/active/0config.cfg.php#L55

Please see where you put the files, if /docroot/, index.php, qcubed.inc.php, project folder, vendor folder etc, then this line must be `define('QCUBED_URL_PREFIX', '');`

There are a couple of projects in my development environment, so I do this: `define('QCUBED_URL_PREFIX', '/first_project');` , for another project `define('QCUBED_URL_PREFIX', '/second_project');` and so on...

In other words, `http://localhost/first_project`, `http://localhost/second_project`, etc ...

Just in case, I'll let you know about such opportunities.

**Comment by kukrik** [comment]
> When I said: "Anybody out there who wants to share projects (code and db structure) which we can study and learn from?" - I was asking the other members of this group if they would share any of their projects so we can learn from them. I am still hoping somebody would.

My project "NestedSortable" is still under development, I will make this project public when it is ready. But there are still small mistakes. If you want to explore this project a bit, I can give you a private link. If you are a Facebook user, search for me by my name Tiit Papp and make a request to join. And write in my private message.

EDIT: Or write to me tiit.papp@gmail.com

**Comment by kukrik** [comment]
If you want, you can also explore and test the plugins I created. Comments and observations are welcome!

https://github.com/kukrik/bootstrap-datetimepicker
https://github.com/kukrik/bootstrap-clockpicker
https://github.com/kukrik/toastr
https://github.com/kukrik/select2
https://github.com/kukrik/counterup2
https://github.com/kukrik/bootstrap
https://github.com/kukrik/ckeditor

These last 2 plugins were not created by me. Since these repositories were not maintained for a long time, I made repos of them and updated them to work.
